public/javascripts/ScrabbleBag.js

```diff
@@ -0,0 +1,75 @@
+const _letterBag = new WeakMap()
+const _keys = new WeakMap()
+const _values = new WeakMap()
+const _distinctValues = new WeakMap()
+const _amountsWithLetters = new WeakMap()
+const _errors = new WeakMap()
+
+class ScrabbleBag {
+
+  constructor() {
+    _letterBag.set(this, {
```

Comment:
Are you using `WeakMap` to create private variables for this class? Thats not a pattern I've ever seen before. While it's fascinating and might have a niche use case, I recommend you consider just storing these values on the instance itself. `this. _letterBag = {…}`